### ISSUE TITLE
[package] autoconf/2.69: cannot be built on FreeBSD

### ISSUE THREAD

**Issue by ericLemanissier** [conan-io/conan-center-index#4014]
As title says, autoconf/2.69: cannot be built on FreeBSD. it fails with `../bin/autom4te: not found`

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **autoconf/2.69**
  * Operating System+version: **FreeBSD** https://github.com/vmactions/freebsd-vm
  * Compiler+version: **clang 10.0**
  * Conan version: **conan 1.32.1**
  * Python version: **Python 3.7**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
CC and CXX: clang, clang++ 
Found clang 10.0
clang>=8, using the major as version
Default settings
	os=FreeBSD
	os_build=FreeBSD
	arch=x86_64
	arch_build=x86_64
	compiler=clang
	compiler.version=10
	compiler.libcxx=libc++
	build_type=Release
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
autoconf/2.69: Configuring sources in /root/.conan/data/autoconf/2.69/_/_/source

autoconf/2.69: Copying sources to build folder
autoconf/2.69: Building your package in /root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a
autoconf/2.69: Generator txt created conanbuildinfo.txt
autoconf/2.69: Calling build()
autoconf/2.69: Calling:
 > /root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/source_subfolder/configure '--datarootdir=/root/.conan/data/autoconf/2.69/_/_/package/bd9ac12e1c28150c93ecb5bd8f852968618b578a/bin/share' '--prefix=/root/.conan/data/autoconf/2.69/_/_/package/bd9ac12e1c28150c93ecb5bd8f852968618b578a' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' 
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/source_subfolder/build-aux/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... nawk
checking whether make sets $(MAKE)... yes
checking build system type... x86_64-unknown-freebsd12.2
checking host system type... x86_64-unknown-freebsd12.2
configure: autobuild project... GNU Autoconf
configure: autobuild revision... 2.69
configure: autobuild hostname... freebsd
configure: autobuild timestamp... 20201229T063857Z
checking whether /bin/sh -n is known to work... no
checking for characters that cannot appear in file names... none
checking whether directories can have trailing spaces... yes
checking for expr... /bin/expr
checking for GNU M4 that supports accurate traces... /root/.conan/data/m4/1.4.18/_/_/package/ae7365f455ceb04fbe77b819f705aaaa46996bda/bin/m4
checking whether /root/.conan/data/m4/1.4.18/_/_/package/ae7365f455ceb04fbe77b819f705aaaa46996bda/bin/m4 accepts --gnu... yes
checking how m4 supports trace files... --debugfile
checking for perl... /usr/local/bin/perl
  echo  &&\
  echo 'm4_define([m4_PACKAGE_NAME],      [GNU Autoconf])' && \
  echo 'm4_define([m4_PACKAGE_TARNAME],   [autoconf])' && \
  echo 'm4_define([m4_PACKAGE_VERSION],   [2.69])' && \
  echo 'm4_define([m4_PACKAGE_STRING],    [GNU Autoconf 2.69])' && \
  echo 'm4_define([m4_PACKAGE_BUGREPORT], [bug-autoconf@gnu.org])' && \
  echo 'm4_define([m4_PACKAGE_URL],       [http://www.gnu.org/software/autoconf/])' && \
  echo 'm4_define([m4_PACKAGE_YEAR],      ['`sed 's/^\([0-9][0-9][0-9][0-9]\).*/\1/;q' /root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/source_subfolder/ChangeLog`'])'; \
} > version.m4-t
mv version.m4-t version.m4
gmake[3]: Leaving directory '/root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/lib/m4sugar'
AUTOM4TE_PERLLIBDIR='/root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/source_subfolder'/lib AUTOM4TE_CFG='../lib/autom4te.cfg'         ../bin/autom4te -B '..'/lib -B '/root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/source_subfolder'/lib         --language M4sh --cache '' --melt /root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/source_subfolder/bin/autoconf.as -o autoconf.in
/bin/sh: ../bin/autom4te: not found
gmake[2]: *** [Makefile:641: autoconf.in] Error 127
gmake[2]: Leaving directory '/root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a/bin'
gmake[1]: *** [Makefile:357: all-recursive] Error 1
gmake[1]: Leaving directory '/root/.conan/data/autoconf/2.69/_/_/build/bd9ac12e1c28150c93ecb5bd8f852968618b578a'
gmake: *** [Makefile:292: all] Error 2
```

</details>

**Comment by ericLemanissier** [comment]
@madebr would you have an idea what could be done here ?

**Comment by prince-chrismc** [comment]
https://www.linuxquestions.org/questions/linux-newbie-8/autpoint-autom4te-not-found-4175571980/#post5498549

**Comment by ericLemanissier** [comment]
I don't think it's the same issue. in my case the autom4te script was built at the step before the failure. In this post the user did not install autoconf.

I added some debug tests in https://github.com/conan-io/conan-center-index/pull/4078:

- `ls -al bin` shows `-r-xr-xr-x  1 root  wheel  32616 Jan  4 17:49 autom4te` so it exists and is executable : https://github.com/ericLemanissier/conan-center-index/runs/1645302890?check_suite_focus=true#step:3:2614
- `cat bin/autom4te` works: https://github.com/ericLemanissier/conan-center-index/runs/1645302890?check_suite_focus=true#step:3:2619
- `./bin/autom4te` fails with `/bin/sh: ./bin/autom4te: not found`: https://github.com/ericLemanissier/conan-center-index/runs/1645302890?check_suite_focus=true#step:3:3681

I'm lost :/

**Comment by ericLemanissier** [comment]
ok, it was an perl setup issue: solved with `ln -s /usr/local/bin/perl /usr/bin/perl`

**Comment by madebr** [comment]
Would a `perl` recipe have helped?

**Comment by ericLemanissier** [comment]
Actually, it's a bug of the recipe. the correct path is identified beforehand : https://github.com/ericLemanissier/conan-center-index/runs/1645302890?check_suite_focus=true#step:3:2510 but because of this hunk https://github.com/conan-io/conan-center-index/blob/master/recipes/autoconf/all/patches/0002-no-perl-path-in-shebang.patch#L11 anther path is used in autom4te shebang : https://github.com/ericLemanissier/conan-center-index/runs/1645302890?check_suite_focus=true#step:3:2600

**Comment by madebr** [comment]
The patch is used because on Windows, the perl path contains build system specific paths.
Maybe the replacement must be os-specific?

**Comment by madebr** [comment]
How about replacing it with `/usr/bin/env perl`?

**Comment by ericLemanissier** [comment]
`/usr/bin/env perl` is appealing !  (if it works)

**Comment by ericLemanissier** [comment]
seems to be fixed in 2.71: https://github.com/ericLemanissier/conan-center-index/runs/3930477401?check_suite_focus=true#step:3:15602
